homeassistant/components/vartastorage/sensor.py

```diff
@@ -0,0 +1,221 @@
+"""Sensor platform of the VARTA Storage integration."""
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from vartastorage import vartastorage
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Entry setup which was forwarded by __init__.py."""
+
+    async def async_update_data():
+        """Fetch data and preo-process the data from API endpoint."""
+
+        def sync_update():
+            """Utilizing synchronous task as the used PyPI Package is not built with async."""
+            varta = vartastorage.VartaStorage(entry.data["host"], entry.data["port"])
+            # Collect all data from the device at once
+            varta.get_all_data()
+            return varta
+
+        try:
```

Comment:
Limit the `try`/`except` block to the result fetch

```suggestion
        try:
            async with async_timeout.timeout(10):
                result = await hass.async_add_executor_job(sync_update)
        except Exception as failed_update:
                raise UpdateFailed("Error communicating with API") from failed_update
        
        thisdict = {}
        ....
```

Reply:
done